src/components/InputName/InputName.tsx

```diff
@@ -3,9 +3,72 @@ import React, { useState } from 'react'
 import styled, { css } from 'styled-components'
 import { fonts } from '../../globalStyles/fonts'
 
+export interface InputProps {
+  label: string
+  placeholder: string
+  inputName?: string
+  inputType: string
+  isValid: boolean
+  prop?: boolean
+}
+
+interface styledPropType {
+  errorMessage: boolean
+}
+
+export const Input: React.FC<InputProps> = ({
+  label,
+  placeholder,
+  inputName,
+  inputType,
+  isValid,
+}) => {
+  const [showPassword, setShowpassword] = useState(false);
+  const [inputValue, setInputValue] = useState('');
+
+  function inputValueHandler(e: string) {
+    setInputValue(e);
+  }
+  return (
+    <InputContainer>
+      <div style={{ position: 'relative' }}>
+      
+        <LabelContainer>
+          <Label>{label}</Label>
+          {!isValid && 
+          <ErrorMessage>
+            {inputName} invalid
+          </ErrorMessage>}
+        </LabelContainer>
+        
+        <InputElement
+          value={inputValue}
+          placeholder={placeholder}
+          type={showPassword ? 'text' : inputType}
+          name={inputName}
+          onChange={(e: any) => inputValueHandler(e.target.value)}
+          errorMessage={!isValid}
+          autoComplete='off'
+          required
+        />
+        {inputName === 'password' ? (
```

Comment:
shouldnt it be `type ==='password'` ?

Reply:
Yes, `type ==='password'` is better. I will change that in my next task. Thanks!